extensions/ql-vscode/src/query-history/query-history-manager.ts

```diff
@@ -573,40 +571,46 @@ export class QueryHistoryManager extends DisposableObject {
     }
   }
 
+  isSuccessfulCompletedLocalQueryInfo(
+    item: QueryHistoryInfo,
+  ): item is CompletedLocalQueryInfo {
+    return item.t === "local" && item.completedQuery?.successful === true;
+  }
+
   async handleCompareWith(
     singleItem: QueryHistoryInfo,
     multiSelect: QueryHistoryInfo[] | undefined,
   ) {
     multiSelect ||= [singleItem];
 
-    try {
-      // local queries only
-      if (singleItem?.t !== "local") {
-        throw new Error("Please select a local query.");
-      }
-
-      if (!singleItem.completedQuery?.successful) {
-        throw new Error(
-          "Please select a query that has completed successfully.",
-        );
-      }
+    if (
+      !this.isSuccessfulCompletedLocalQueryInfo(singleItem) ||
+      !multiSelect.every(this.isSuccessfulCompletedLocalQueryInfo)
+    ) {
+      throw new Error(
+        "Please only select local queries that have completed successfully.",
+      );
+    }
 
-      const from = this.compareWithItem || singleItem;
-      const to = await this.findOtherQueryToCompare(from, multiSelect);
+    const fromItem =
```

Comment:
(Optional) I found this a little bit hard to read (because of the conditional). Would it make sense to extract that out or perhaps the whole statement could be extracted as a function?